x-pack/legacy/plugins/siem/public/components/ml_popover/helpers.tsx

```diff
@@ -4,98 +4,57 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { ConfigTemplate, IndexPatternSavedObject, Job } from './types';
+import { SiemJob } from './types';
 
 /**
- * Returns all `jobIds` for each configTemplate provided
- *
- * @param templates ConfigTemplates as provided by ML Team (https://github.com/elastic/machine-learning-data/issues/194#issuecomment-505779406)
- */
-export const getJobsToInstall = (templates: ConfigTemplate[]): string[] =>
-  templates.reduce((jobs: string[], template) => [...jobs, ...template.jobs], []);
-
-/**
- * Returns which ConfigTemplates that need to be installed based off of which Jobs are already installed and the configured indexPattern
- *
- * @param templates ConfigTemplates as provided by ML Team
- * @param installedJobIds list of installed JobIds
- * @param indexPatterns list of the user's currently configured IndexPatterns
- */
-export const getConfigTemplatesToInstall = (
-  templates: ConfigTemplate[],
-  installedJobIds: string[],
-  indexPatterns: string[]
-): ConfigTemplate[] =>
-  templates
-    .filter(ct => !ct.jobs.every(ctJobId => installedJobIds.includes(ctJobId)))
-    .filter(ct => indexPatterns.includes(ct.defaultIndexPattern));
-
-/**
- * Returns a filtered array of Jobs that based on filterGroup selection (Elastic vs Custom Jobs) and any user provided filterQuery
+ * Returns a filtered array of Jobs according to JobsTableFilters selections
  *
  * @param jobs to filter
- * @param embeddedJobIds jobIds as defined in the ConfigTemplates provided by the ML Team
+ * @param selectedGroups groups to filter on
  * @param showCustomJobs whether or not to show all Custom Jobs (Non-embedded Jobs in SIEM Group)
  * @param showElasticJobs whether or not to show Elastic Jobs (Embedded ConfigTemplate Jobs)
  * @param filterQuery user-provided search string to filter for occurrence in job names/description
  */
-export const getJobsToDisplay = (
-  jobs: Job[] | null,
-  embeddedJobIds: string[],
-  showCustomJobs: boolean,
-  showElasticJobs: boolean,
-  filterQuery?: string
-): Job[] =>
-  jobs
-    ? searchFilter(
-        jobs
-          .filter(job => (showCustomJobs ? embeddedJobIds.includes(job.id) : true))
-          .filter(job => (showElasticJobs ? !embeddedJobIds.includes(job.id) : true)),
-        filterQuery
-      )
-    : [];
+export const filterJobs = ({
+  jobs,
+  selectedGroups,
+  showCustomJobs,
+  showElasticJobs,
+  filterQuery,
+}: {
+  jobs: SiemJob[];
+  selectedGroups: string[];
+  showCustomJobs: boolean;
+  showElasticJobs: boolean;
+  filterQuery: string;
+}): SiemJob[] =>
+  searchFilter(
+    jobs
+      .filter(job => !showCustomJobs || (showCustomJobs && !job.isElasticJob))
+      .filter(job => !showElasticJobs || (showElasticJobs && job.isElasticJob))
+      .filter(
+        job => selectedGroups.length === 0 || selectedGroups.some(g => job.groups.includes(g))
```

Comment:
You wan to know a funny trick?

You can do this:

```ts
selectedGroups.some(job.groups.includes))
```

And since you have only one argument it automatically should pass it to includes. Some people don't like that trick though and others embrace it like crazy.

No changes requested, just pointing it out.

Reply:
Oh yeah! I knew of that feature of `some()` (as with a few of the other array methods) but hadn't made the syntactic sugary connection before. Thanks for sharing buddy! 🙂

Reply:
This little piece of 🍬 resulted in the crash outlined here -- https://github.com/elastic/kibana/pull/48067#issuecomment-542032835, reverting.

Reply:
haha, sorry about that bad advice. adding strike throughs above.